File: Polynomial Regression/polynomial_regression.R
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]


#Fitting the dataset in the Linear Regression model
lin_reg = lm(formula = Salary ~ .,
             data = dataset)

#Fittng the dataset in the Polynomial Regression model
dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4
poly_reg = lm(formula = Salary ~ .,
              data = dataset)


library(ggplot2)
ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(lin_reg, newdata = dataset)),
            color = 'blue') +
  ggtitle('Truth and bluff(Linear Regression)') +
  xlab('Level') +
  ylab('Salary')


library(ggplot2)
ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(poly_reg, newdata = dataset)),
            color = 'blue') +
  ggtitle('Truth and bluff(Linear Regression)') +
  xlab('Level') +
  ylab('Salary')

y_pred = predict(lin_reg, data.frame(Level=6.5))

y_pred_new = predict(poly_reg, data.frame(Level=6.5,
                                          Level2=6.5^2,
                                          Level3=6.5^3,
                                          Level4=6.5^4))

File: Regression models/Polynomial Regression/polynomial_regression.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Position_Salaries.csv')

X = dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values

from sklearn.linear_model import LinearRegression

lin_reg = LinearRegression()
lin_reg.fit(X,y)

#Fitting the polynomial to dataset
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 4)
X_poly = poly_reg.fit_transform(X)

lin_reg_2 = LinearRegression()
lin_reg_2.fit(X_poly,y)

#Visualinsing the Linear Model
plt.scatter(X,y,color = 'red')
plt.plot(X, lin_reg.predict(X),color='blue')
plt.title('Truth or Bluff(Linear Regression)')
plt.xlabel('Position levels')
plt.ylabel('Salary')
plt.show()


#Visualinsing the Polynomial Model
X_grid = np.arange(min(X),max(X),0.1)
X_grid = X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,color = 'red')
plt.plot(X, lin_reg_2.predict(poly_reg.fit_transform(X)),color='blue')
plt.title('Truth or Bluff(Polynomial Regression)')
plt.xlabel('Position levels')
plt.ylabel('Salary')
plt.show()

#Predicting the result using Linear Regression
lin_reg.predict(6.5)

#Predicting the result using Polynomial Regression
lin_reg_2.predict(poly_reg.fit_transform(6.5))
